layer/layer_store.go

```diff
@@ -899,6 +918,74 @@ func (ls *layerStore) deleteUnreferencedDriverLayers(ids []string) int {
 	return total
 }
 
+func (ls *layerStore) findUnusedOverlayFiles() ([]string, error) {
+	var fileNameOverlayIdRegex = regexp.MustCompile("^([a-z0-9]+).*")
+	var overlaysDir = "/var/lib/docker/overlay2"
```

Comment:
Is the problem only with the overlay2 storage engine, or are we only addressing it on overlay2?

Reply:
It might be a problem with `aufs` as well, but Shaun specifically requested `overlay2`, and the fix in engine-check is actually different depending on which you are using.

So I only implemented `overlay2` since we are moving all devices to overlay for balenaOS 3.0.